From d9bc718871810d18c70ba9d4e9ee23071bd6db36 Mon Sep 17 00:00:00 2001
From: Stefan Berger <stefanb@linux.vnet.ibm.com>
Date: Tue, 21 Jul 2020 18:39:07 -0400
Subject: [PATCH 2/2] build-sys: Explicitly link libswtpm_libtpms with -lcrypto
 (Gentoo)

This patch fixes the following linker issue reported for Gentoo in
issue #280.

ld.lld: error: /var/tmp/portage/app-crypt/swtpm-0.3.1-r1/work/swtpm-0.3.1/src/swtpm/.libs/libswtpm_libtpms.so: undefined reference to EVP_sha512
ld.lld: error: /var/tmp/portage/app-crypt/swtpm-0.3.1-r1/work/swtpm-0.3.1/src/swtpm/.libs/libswtpm_libtpms.so: undefined reference to PKCS5_PBKDF2_HMAC
ld.lld: error: /var/tmp/portage/app-crypt/swtpm-0.3.1-r1/work/swtpm-0.3.1/src/swtpm/.libs/libswtpm_libtpms.so: undefined reference to SHA512

Signed-off-by: Stefan Berger <stefanb@linux.ibm.com>
---
 configure.ac          | 2 ++
 src/swtpm/Makefile.am | 3 ++-
 2 files changed, 4 insertions(+), 1 deletion(-)

diff --git a/configure.ac b/configure.ac
index 56d3f09..7aaa201 100644
--- a/configure.ac
+++ b/configure.ac
@@ -150,6 +150,8 @@ openssl)
 	AC_CHECK_HEADERS([openssl/aes.h],[],
 			 AC_MSG_ERROR(Is openssl-devel/libssl-dev installed?))
 	AC_MSG_RESULT([Building with openssl crypto library])
+	LIBCRYPTO_LIBS=$(pkg-config --libs libcrypto)
+	AC_SUBST([LIBCRYPTO_LIBS])
 	;;
 esac
 
diff --git a/src/swtpm/Makefile.am b/src/swtpm/Makefile.am
index dd2a63a..b5f15e3 100644
--- a/src/swtpm/Makefile.am
+++ b/src/swtpm/Makefile.am
@@ -69,7 +69,8 @@ libswtpm_libtpms_la_LIBADD = \
 	$(LIBTPMS_LIBS) \
 	$(GLIB_LIBS) \
 	$(LIBRT_LIBS) \
-	$(LIBSECCOMP_LIBS)
+	$(LIBSECCOMP_LIBS) \
+	$(LIBCRYPTO_LIBS)
 
 bin_PROGRAMS = swtpm
 if WITH_CUSE
-- 
2.26.2

